### ISSUE TITLE
even example doesn't compile on arduino 1.6.6

### ISSUE THREAD

**Issue by NitrofMtl** [nomadnt/uHTTP#12]
Hi.

Hi try to use uHTTP but even when I load the example, it doesn't compile and get a lot of error, those are:

[code]

Build options changed, rebuilding all
C:\Users\o_for\Documents\Arduino\libraries\uHTTP-master\uHTTP.cpp: In member function 'EthernetClient uHTTP::available()':
C:\Users\o_for\Documents\Arduino\libraries\uHTTP-master\uHTTP.cpp:84:52: error: 'strncmp_P' was not declared in this scope
                  if(strncmp_P(buffer, PSTR("OP"), 2) == 0) __method = uHTTP_METHOD_OPTIONS;
                                                    ^
C:\Users\o_for\Documents\Arduino\libraries\uHTTP-master\uHTTP.cpp:149:56: error: 'strncmp_P' was not declared in this scope
                    if(strncmp_P(buffer, PSTR("Auth"), 4) == 0) header = AUTHORIZATION;
                                                        ^
C:\Users\o_for\Documents\Arduino\libraries\uHTTP-master\uHTTP.cpp: In member function 'const char* uHTTP::parse(const char*, char*, const __FlashStringHelper*)':
C:\Users\o_for\Documents\Arduino\libraries\uHTTP-master\uHTTP.cpp:350:56: error: 'strtok_rP' was not declared in this scope
   sub = strtok_rP(act, (const PROGMEM char *) sep, &ptr);
                                                        ^
C:\Users\o_for\Documents\Arduino\libraries\uHTTP-master\uHTTP.cpp: In member function 'const char* uHTTP::parse(const __FlashStringHelper*, char*, const __FlashStringHelper*)':
C:\Users\o_for\Documents\Arduino\libraries\uHTTP-master\uHTTP.cpp:368:105: error: 'strncmp_P' was not declared in this scope
  for(act = buffer; strncmp_P(sub, (const PROGMEM char *) needle, strlen_P((const PROGMEM char *) needle)); act = NULL){
                                                                                                         ^
C:\Users\o_for\Documents\Arduino\libraries\uHTTP-master\uHTTP.cpp:369:56: error: 'strtok_rP' was not declared in this scope
   sub = strtok_rP(act, (const PROGMEM char *) sep, &ptr);
                                                        ^
Multiple libraries were found for "Ethernet.h"
 Used: C:\Users\o_for\Documents\Arduino\libraries\Ethernet
 Not used: C:\arduino-nightly\libraries\Ethernet
Error compiling.

[/code]

**Comment by NitrofMtl** [comment]
Correction: I t seem some example works other don't; on parse, rest_interface and sd_doc_root example, it compile on arduino UNO but not a Arduino DUE. for the base example, it complitly froze on compile on UNO as well but with the eror:  


Exception in thread "Thread-639" java.lang.StackOverflowError
	at java.lang.Character.codePointAt(Character.java:4866)
	at java.util.regex.Pattern$CharProperty.match(Pattern.java:3775)
	at java.util.regex.Pattern$Branch.match(Pattern.java:4604)
	at java.util.regex.Pattern$GroupHead.match(Pattern.java:4658)
	at java.util.regex.Pattern$Loop.match(Pattern.java:4785)
	at java.util.regex.Pattern$GroupTail.match(Pattern.java:4717)



and a lot of line of "at java..."

**Comment by hreintke** [comment]
@NitrofMtl : 
It seems that you are not using the Sming environment but the Arduino.
Sming is used for esp8266 programming using a cross compile on windows/unix/mac. 
You can find installation instructions on the "Getting started" part of the repository.
In the repository are also several examples on the usage of sming

**Comment by NitrofMtl** [comment]
ok.. but this lib have been wrote for arduino... working on UNO but not DUE board... I've try to instal swing anyway... doesn't fix the problem

**Comment by hreintke** [comment]
Arduino libraries only run on sming after being adapted to the sming environment.
There are several libraries included in the sming repository. You can check there whether "your" library has a sming version. I not they can be used as example for sming adaption.
If ready and working, you can create a pull request to include the new library into sming.

**Comment by hreintke** [comment]
@NitrofMtl : 
Sorry but I replied to the issue in the assumption it was a question on the sming repository.
Only now I see it is in the uHTTP repository.
Forget my inital comments, this is the answer on the original issue.

I worked shortly with uHTTP when starting with Arduino. After that I discontinued that and have  no further experience with uHTTP. Can't help you with your compile issues.

**Comment by NitrofMtl** [comment]
Ok. It appear that the problem came from that Arduino DUE have a lot more memory and all PROGMEM function should be remove. I din't have the skills to do it, maybe try something... :P

**Comment by NitrofMtl** [comment]
Ok. upgraded my skills apparently :P 
I was effectevely the PROGMEM function that have to be removed. so now it work.
Also, I tricked some function of my own to ease some basic function, like reconise a file on sd card, and send it to client ans some other function of that kind. 

I will probably post it soon



see ya.

**Comment by tonyputi** [comment]
Hi, can I close this issue?

**Comment by NitrofMtl** [comment]
yep. find it hard to follow to do a pull request so maybe should I do it here right now. I work on modifying the library to work with Arduino DUE and add some more feature like sdFile request.
Not completed yet but still most of it work. [uHTTP Nitrof version](https://github.com/NitrofMtl/uHTTP)

**Comment by tonyputi** [comment]
Ok. If you want I could add you on uHTTP develop team.
